crates/burn-core/src/nn/rope_encoding.rs

```diff
@@ -0,0 +1,216 @@
+use crate as burn;
+use crate::config::Config;
+use crate::module::Module;
+use crate::tensor::backend::Backend;
+use crate::tensor::Tensor;
+use alloc::vec;
+use burn_tensor::Int;
+
+#[cfg(not(feature = "std"))]
+use num_traits::Float;
+
+/// Configuration to create a [RotaryEncoding](RotaryEncoding) layer.
+#[derive(Config, Debug)]
+pub struct RotaryEncodingConfig {
+    /// Maximum sequence length of input
+    max_sequence_length: usize,
+
+    /// Size of the input embedding or hidden dimension
+    d_model: usize,
+
+    /// Scaling factor for frequency computation. Defaults to 10000.0
+    #[config(default = "10000.0")]
+    theta: f32,
+}
+
+impl RotaryEncodingConfig {
+    /// Initialize a new [RotaryEncoding](RotaryEncoding) module.
+    ///
+    /// Panics if the size of input embedding dimension is not even.
```

Comment:
Can also mention theta needing to be positive

Reply:
Thanks @dcvz, added this as well.